Themes: add 'Using SCSS' guide (#5286)
DevExpress/devextreme-documentation: concepts/60 Themes and Styles/05 Predefined Themes/50 Apply a Theme/05 Using SCSS.md

> You can import the theme bundle from the `bundles` folder from the `devextreme` package in the root `.scss` file:

Reviewer: First, I'm going to question your article usage in this scenario:

> You can import **the** theme bundle from ...

What theme bundle? You haven't declared it anywhere else prior to this point. This is the very first line in this topic.

Second... From from

> You can import the theme bundle **from** the `bundles` folder **from** the `devextreme` package.

On possible alternative is "in", however, this change would leave us with "in in" instead of "from from", which is hardly better. However, there is a way to resolve this scenario. There's always a way.

```suggestion
You can import a theme bundle from the `bundles` folder in the `devextreme` package found in the root `.scss` file:
```

OR (there are different interpretations) -

```suggestion
You can import a theme bundle from the `bundles` folder of the `devextreme` package in the root `.scss` file:
```

OR maybe you'll be able to come up with something way better than that altogether...

    /* @use 'devextreme/scss/bundles/dx.theme_name.scss';  */
    @use 'devextreme/scss/bundles/dx.carmine.scss';

> Or you can import individual files for the whole theme. The example below imports files of the `Material Blue Light Compact` theme

Reviewer: > Or you can import individual files for **the** whole theme.

Again, I see a definite article. Why? We don't have to speak of a specific theme, or even **the same** indefinite theme we mentioned earlier on. Because the things we speak of are valid for **any** possible theme. Therefore, there is no reason to interpret this case as definite.

```suggestion
Alternatively, you can import individual files that comprise a theme. The example below imports files from the `Material Blue Light Compact` theme:
```

> Alternatively, you can import individual files that comprise a theme. The theme folder can be either `material` or `generic` depending on the theme family. The example below imports files from the `Material Blue Light Compact` theme:

Reviewer: ```suggestion
Alternatively, you can import individual files that comprise a theme. The theme folder name is either `material` or `generic` depending on the theme family. The example below imports files from the `Material Blue Light Compact` theme:
```

    @use 'devextreme/scss/widgets/material/colors' with ($color: 'blue', $mode: 'light');
    @use 'devextreme/scss/widgets/material/sizes' with ($size: 'compact');
    @use 'devextreme/scss/widgets/material/icons';
    @use 'devextreme/scss/widgets/material/widget';
    @use 'devextreme/scss/widgets/material/card';
    @use 'devextreme/scss/widgets/material/fieldset';
    @use 'devextreme/scss/widgets/material/common';
    @use 'devextreme/scss/widgets/base/resizable';
    @use 'devextreme/scss/widgets/base/draggable';
    @use 'devextreme/scss/widgets/base/ui';
    @use 'devextreme/scss/widgets/material';

> To import only specific components, replace `@use 'devextreme/scss/widgets/material';` with `@use 'devextreme/scss/widgets/componentName';`:

Reviewer 2: a minor adjustment:

replace `@use 'devextreme/scss/widgets/themeName';` with `@use 'devextreme/scss/widgets/themeName/componentName';`:

> The theme path can be either `material` or `generic` depending on the theme family. The second solution is useful if you need to import only specific components. To do this, replace `@use 'devextreme/scss/widgets/material';` with `@use 'devextreme/scss/widgets/componentName';`:

Reviewer: > The theme path **can be** either `material` or `generic` depending on the theme family.

But that's not quite true. `material` or `generic` is only a part of the path, not the entire path itself.

> The second solution is useful if you need to import only specific components.

What "second solution" you are referring here to? I do not understand. If you meant this

> Alternatively, you can import individual files that comprise a theme.

You mentioned it way too far away from this point; therefore, it's impossible to connect it with the current context.

> The second solution is useful if you need to import only specific components. **To do this**, replace `@use 'devextreme/scss/widgets/material';` with `@use 'devextreme/scss/widgets/componentName';`:

Do what? This entire paragraph feels a bit out-of-place here. We definitely need to do something about it.

    @use "devextreme/scss/widgets/generic/colors" with ($color: "darkmoon");
    @use "devextreme/scss/widgets/generic/sizes" with ($size: "default");
    @use "devextreme/scss/widgets/generic/typography";
    @use "devextreme/scss/widgets/generic/icons";
    @use "devextreme/scss/widgets/generic/widget";
    @use "devextreme/scss/widgets/generic/card";
    @use "devextreme/scss/widgets/generic/fieldset";
    @use "devextreme/scss/widgets/generic/common";
    @use "devextreme/scss/widgets/base/resizable";
    @use "devextreme/scss/widgets/base/draggable";
    @use "devextreme/scss/widgets/base/ui";
    @use "devextreme/scss/widgets/generic/textBox";
    @use "devextreme/scss/widgets/generic/button";

> You can use the following variables in the imports:

Reviewer: > You can use the following variables in **the imports**:

First, I'm going to question your article usage.

Second, I'm going to ask what "imports" is? What is that noun "import"? Import is a verb, not a noun. The only noun-like usage of "imports" is "imported goods". It's unlikely we can interpret imported files as such:

```suggestion
You can use the following variables in imported files:
```

- `$mode`    

> The theme main palette (`dark` or `light`). This variable has effect for `Material` themes only.

Reviewer: > The theme main palette

Something is amiss with the word order here.

> This variable has effect **for**

This is strange preposition usage. Something has effect **on** something, not **for** something.

```suggestion
The main theme palette (`dark` or `light`). This variable has effect on `Material` themes only.
```

- `$color`    

> Derived from the theme name. Represents the theme's main color (blue, carmine, contrast, etc.)

Reviewer: Represents. I never thought I'd see it again at some point...

https://tw.devexpress.devx/guidelines/language-guidelines/misused-words/represent/

Under no circumstances "represents" should end up in our documentation.

```suggestion
Derived from the theme name. This variable defines the main theme color (blue, carmine, contrast, etc.)
```

- `$size:`    
Can be `compact` or `default`.

> [note] If you want to customize inner variables (like components fonts, color, etc.), examine the file content inside the `devextreme/scss/widgets/` folder to learn what variables can be customized. Note that this code is internal and DevExtreme does not have a complete documentation for the SCSS structure. Variable names, order of import, and the folder structure may be changed without any warning. To customize specific variables, use [ThemeBuilder](/Documentation/Guide/Themes_and_Styles/ThemeBuilder/) or override [individual CSS classes](/Documentation/Guide/Themes_and_Styles/HTML-Based_Components_Customization/#Individual_CSS_Classes).

Reviewer: > like component**s** fonts

Agreement of numbers. You wanted to say "fonts of components". However, a more compact form would look only as follows:

> component font**s**

Only the last word is a noun, and only this noun can be plural.

> examine **the** file content inside

The word "content" is uncountable. Is this case definite? No. We don't know anything of this content in this context, and it's entirely unimportant.

> **a** complete documentation

Like "text", documentation is uncountable. While it's possible to interpret "documentation" as a piece of documentation - a documentation repository, a documentation topic, etc. - this is too far-fetched in this context.

```suggestion
[note] If you want to customize internal variables (like component fonts, color, etc.), examine file content inside the `devextreme/scss/widgets/` folder to learn about customizable variables. Note that this code is internal and DevExtreme does not have complete documentation for the SCSS structure. Variable names, import order, and the folder structure may change in future builds without official notice. To customize specific variables, use [ThemeBuilder](/Documentation/Guide/Themes_and_Styles/ThemeBuilder/) or override [individual CSS classes](/Documentation/Guide/Themes_and_Styles/HTML-Based_Components_Customization/#Individual_CSS_Classes).
```
